### ISSUE TITLE
m4 and Makefiles not automatically recognized

### ISSUE THREAD

**Issue by darealshinji** [notepadqq/notepadqq#694]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/notepadqq/notepadqq/blob/master/CONTRIBUTING.md#question
```

**Current behavior**
Opening a \*.m4, \*.ac. \*.mak, Makefile.\* or Makefile doesn't automatically set the language to m4 or Makfile. Previous versions did that. m4 and Makefile syntax came from a patch that I once submitted so I guess the auto-detection got lost at a CodeMirror update?

**Expected behavior**
See "Current behavior".

**Minimal reproduction of the problem with instructions**
Opening \*.m4, \*.ac. \*.mak, Makefile.\* or Makefile.

**Please tell us about your environment:**
OS: Ubuntu 18.04
Nqq: v1.4.8 (snap package version "1.4.8+git"  rev845 from edge channel)
Qt: 5.5.1

**Comment by Teklad** [comment]
Possibly a regex issue after moving detection to cxx but its hard to tell.
I'll look into it.

**Comment by JuBan1** [comment]
It seems all the needed patches are still in the CM directory (mode folders and entries in meta.js are present). So I'd hazard the guess that it's on Nqq's side too.

**Comment by JuBan1** [comment]
Fixed in a6293bb. We've had a bug in our name matching.
